src/ks-clock-face.c

```diff
@@ -116,82 +118,113 @@ static void update_proc(Layer *layer, GContext *ctx) {
+  s_radius = prv_anim_percentage(dist_normalized, FINAL_RADIUS);
 
-  s_canvas_layer = layer_create(window_bounds);
-  layer_set_update_proc(s_canvas_layer, update_proc);
-  layer_add_child(window_layer, s_canvas_layer);
+  layer_mark_dirty(s_canvas_layer);
 }
 
-static void window_unload(Window *window) {
-  layer_destroy(s_canvas_layer);
+static void prv_hands_update(Animation *anim, AnimationProgress dist_normalized) {
+  s_anim_time.hours = prv_anim_percentage(dist_normalized, prv_hours_to_minutes(s_last_time.hours));
+  s_anim_time.minutes = prv_anim_percentage(dist_normalized, s_last_time.minutes);
+
+  layer_mark_dirty(s_canvas_layer);
 }
 
-/*********************************** App **************************************/
+static void prv_start_animation() {
+  // Prepare animations
+  static AnimationImplementation s_radius_impl = {
+    .update = prv_radius_update
+  };
+  prv_animate(ANIMATION_DURATION, ANIMATION_DELAY, &s_radius_impl, false);
 
-static int anim_percentage(AnimationProgress dist_normalized, int max) {
-  return (int)(float)(((float)dist_normalized / (float)ANIMATION_NORMALIZED_MAX) * (float)max);
+  static AnimationImplementation s_hands_impl = {
+    .update = prv_hands_update
+  };
+  prv_animate(2 * ANIMATION_DURATION, ANIMATION_DELAY, &s_hands_impl, true);
 }
 
-static void radius_update(Animation *anim, AnimationProgress dist_normalized) {
-  s_radius = anim_percentage(dist_normalized, FINAL_RADIUS);
+static void prv_create_canvas() {
+  Layer *window_layer = window_get_root_layer(s_main_window);
+  GRect window_bounds = layer_get_unobstructed_bounds(window_layer);
+  s_canvas_layer = layer_create(window_bounds);
+  layer_set_update_proc(s_canvas_layer, prv_update_proc);
+  layer_add_child(window_layer, s_canvas_layer);
+}
 
-  layer_mark_dirty(s_canvas_layer);
+/*********************************** App **************************************/
+
+// Event fires once, before the obstruction appears or disappears
+static void prv_unobstructed_will_change(GRect final_unobstructed_screen_area, void *context) {
+  if(s_animating) {
+    return;
+  }
+  // Reset the clock animation
+  s_radius = 0;
+  s_anim_hours_60 = 0;
+}
+
+// Event fires once, after obstruction appears or disappears
+static void prv_unobstructed_did_change(void *context) {
+  if(s_animating) {
+    return;
+  }
+  // Play the clock animation
+  prv_start_animation();
 }
 
-static void hands_update(Animation *anim, AnimationProgress dist_normalized) {
-  s_anim_time.hours = anim_percentage(dist_normalized, hours_to_minutes(s_last_time.hours));
-  s_anim_time.minutes = anim_percentage(dist_normalized, s_last_time.minutes);
+static void prv_window_load(Window *window) {
+  prv_create_canvas();
 
-  layer_mark_dirty(s_canvas_layer);
+  prv_start_animation();
+
+  tick_timer_service_subscribe(MINUTE_UNIT, prv_tick_handler);
+
+  // Subscribe to the unobstructed area events
+  UnobstructedAreaHandlers handlers = {
```

Comment:
I'm not certain I'm totally sold on the current Timeline Peek behaviour - redrawing the animation from a point feels a little jarring when Peek is enabled / disabled.